src/proto/single/utils.rs

```diff
@@ -30,6 +30,19 @@ where
     serializer.serialize_u64(secs)
 }
 
+pub(crate) fn ser_optional_duration<S>(
```

Comment:
```suggestion
pub(crate) fn ser_optional_duration_in_seconds<S>(
```
the "in seconds" part is the main bit that requires this helper function. ditto for `deser`.